src/EigenVector.cpp

```diff
@@ -32,6 +35,41 @@ void EigenVector::write_eigen_vector(const std::string &filename,
   }
 }
 
+bool EigenVector::test_trace_sum(const ssize_t t, const bool do_throw) const {
+  bool fail = false;
+  Eigen::MatrixXcd VdV = V[t].adjoint() * V[t];
+  const std::complex<double> trace = VdV.trace();
+  const std::complex<double> sum = VdV.sum();
+  // we allow for some deviation
+  if( abs( trace.real() - (double)V[t].cols() ) > (double)V[t].cols()*DBL_EPSILON ){
```

Comment:
@pittlerf if the latest commit works, can you please try whether the vdaggerv test passes now? I think the reason we had these problems was an automatic type conversion issue of what is returned by ```V[t].cols()```, resulting in the whole thing being an integer. (note also the compiler warnings that you had in your printing of the difference)